agent/connect/ca/provider_vault.go

```diff
@@ -75,6 +76,12 @@ func (v *VaultProvider) Configure(cfg ProviderConfig) error {
 	}
 
 	client.SetToken(config.Token)
+
+	// We don't want to set the namespace if it's empty to prevent potential
+	// unknown behavior. It's also what the Vault client does.
```

Comment:
For my own curiosity? What kind of "unknown" behaviors?

Also, would you mind expanding on "It's also what the Vault client does." ?

Reply:
I don't know what the behavior is in Vault if a namespace header is set but empty.

The behavior in the Vault API is to make sure the Namespace is empty before setting it: This part is the API Client getting the namespace from the environment: https://github.com/hashicorp/vault/blob/e24037fa8f713f884817e8bec30023e56af0d5cc/api/client.go#L525

This is the behavior in the SSH Agent: https://github.com/hashicorp/vault/blob/e24037fa8f713f884817e8bec30023e56af0d5cc/api/ssh_agent.go#L128

Considering it's validated as a non-empty string in the API code consuming it automatically I figured it'd be better safe than sorry.

Reply:
I've changed the comment which I'm hoping will clarify this a bit